### PR TITLE
Frontenac integration

### PR DESCRIPTION
**Pull request by AndrewLofts**

### CODE REVIEW

File: cPRONTO/oppni.py [hunk 1]
```diff
@@ -20,12 +20,31 @@
 from collections import OrderedDict
 from copy import copy
 from distutils.spawn import find_executable
-from shutil import which
+#from shutil import which -adlofts
```
Comment (raamana) [inline cPRONTO/oppni.py]: delete

File: cPRONTO/oppni.py [hunk 2]
```diff
@@ -846,9 +869,6 @@ def parse_args_check():
     os.environ["PIPELINE_NUMBER_OF_CORES"] = str(options.numcores)
     os.environ["FSLOUTPUTTYPE"] = "NIFTI"
 
-    print("Chosen options: ")
-    print(options)
-
```
Comment (raamana) [inline cPRONTO/oppni.py]: please add back it in

File: cPRONTO/oppni.py [hunk 3]
```diff
@@ -1044,6 +1065,10 @@ def get_hpc_spec(h_type=None, options=None):
         else:
             header.append('{0} {1}{2}'.format(prefix, val[0], val[1]))
 
+    if h_type in ('FRONTENAC', 'SLURM'):
+        # controlling the output job name
+        header.append('{0} --output=%x_%j.log'.format(prefix))
+
```
Comment (raamana) [inline cPRONTO/oppni.py:1073]: this might be a redundant

File: cPRONTO/oppni.py [hunk 4]
```diff
@@ -1118,12 +1143,11 @@ def update_status_and_exit(out_dir):
     try:
         prev_proc_status, prev_options, prev_input_file_all, \
         failed_sub_file, failed_spnorm_file, all_subjects = update_proc_status(out_dir)
-
         try:
             estimate_processing_times(prev_input_file_all, prev_options, all_subjects)
         except:
             print('Processing time could not be estimated.')
-
+        print(prev_proc_status.all_done)
```
Comment (raamana) [inline cPRONTO/oppni.py]: print("All done flag: ".format(prev_proc_status.all_done))

File: cPRONTO/oppni.py [hunk 5]
```diff
@@ -1253,7 +1276,15 @@ def make_job_file_and_1linecmd(file_path):
         hpc_directives.append('{0} {1} {2}'.format(hpc['prefix'], hpc['spec']['workdir'], os.path.dirname(file_path)))
     else:
         # for jobs to run locally, no hpc directives are needed.
+
         hpc_directives.append('#!/bin/bash')
+
+        #add adlofts Jun 26 2018
+        #must add a directoty change so that the file can be found using MATLAB m file loading scheme
+        #TODO : adjust so that cd only occurs for matlab, octave doesnt need, will help with docker images
+        #if environment.lower() in ('matlab'):
```
Comment (raamana) [inline cPRONTO/oppni.py]: clean up or finalize!

Atleast document the need and interactions...

File: cPRONTO/oppni.py [hunk 6]
```diff
@@ -1297,9 +1328,18 @@ def local_exec(script_path):
...
+    script_path = list(script_path)
+    script_path = script_path[0]
     st = os.stat(script_path)
+    #st = os.stat(script_path)            #replaced with lines above
+    print('Check the log file for progress:')
+    print(script_path + '.log')
+
     os.chmod(script_path, st.st_mode | stat.S_IXGRP)
 
+    print('B')
```
Comment (raamana) [inline cPRONTO/oppni.py]: remove!

File: cPRONTO/oppni.py [hunk 7]
```diff
@@ -1297,9 +1328,18 @@ def local_exec(script_path):
     # logger = logging.getLogger(script_path + '.log')
 
     # make it executable
+
+    #adlofts add jun 26 2018
+    script_path = list(script_path)
```
Comment (raamana) [inline cPRONTO/oppni.py]: explain why this must be done.

File: cPRONTO/oppni.py [hunk 8]
```diff
@@ -1611,15 +1670,28 @@ def run_jobs(job_paths, run_locally, num_procs, depends_on_step):
             # if ret_code < 0 or ret_code > 0:
             #     raise OSError('Error executing the {} job locally.'.format(step_id))
 
-            # list of all script paths
-            jobs_list = job_paths.values()
+            #removed Jun 26 2018
```
Comment (raamana) [inline cPRONTO/oppni.py]: delete

File: cPRONTO/oppni.py [hunk 9]
```diff
@@ -1611,15 +1670,28 @@ def run_jobs(job_paths, run_locally, num_procs, depends_on_step):
...
+            #jobs_list = job_paths.values()
 
+            #added Jun 26 2018
+            jobs_list = list(job_paths.values())
```
Comment (raamana) [inline cPRONTO/oppni.py]: lets look into this - this must be a list

File: cPRONTO/oppni.py [hunk 10]
```diff
@@ -1870,7 +1942,7 @@ def submit_jobs():
         if options.run_locally is True and (status_sp is False or status_sp is None):
             raise Exception('Spatial normalization - steps 2 and later failed.')
 
-    # group mask
+    #print(proc_status.gmask)
```
Comment (raamana) [inline cPRONTO/oppni.py]: remove or change to print("group mask processing status: ".format*proc_status.gmask))

File: extra/Pipeline_QC1.m [hunk 11]
```diff
@@ -676,6 +676,7 @@ function Pipeline_QC1( inputfile, varargin )
 out_results = fullfile(QC1_folder,'output_qc1.mat');
 save(out_results,'output_qc1');
 
+disp('OPPNI QC1 CODE COMPLETE');
```
Comment (raamana) [inline extra/Pipeline_QC1.m]: make this identifier bit more unique e.g. OPPNI__STEP__COMPLETION__CODE

**Review by raamana — CHANGES_REQUESTED** [review]

File: cPRONTO/oppni.py (hunk 9 above)
Comment (AndrewLofts) [reply cPRONTO/oppni.py]: inside run_jobs(job_paths...) called from line 1620 = run_jobs(jobs_dict..) with dictionary passed in

File: cPRONTO/oppni.py (hunk 7 above)
Comment (AndrewLofts) [reply cPRONTO/oppni.py]: I think this is redundant after the correction done at 1677, unless at jobs_list[idx_beg:idx_beg + num_procs] changes it to dictionary. jobs_list is indeed a list.

**Review by raamana — APPROVED** [review]

**Pull request merged**